lib/feedx/format/parquet.rb

```diff
@@ -0,0 +1,102 @@
+require 'parquet'
+require 'tmpdir'
+
+class Feedx::Format::Parquet < Feedx::Format::Abstract
+  class Record < Arrow::Record
+    def each_pair
+      container.columns.each do |col|
+        yield col.name, col[index]
+      end
+    end
+  end
+
+  class Decoder < Feedx::Format::Abstract::Decoder
+    def initialize(io, **)
+      super(io)
+
+      @table  = read_table
+      @cursor = 0
+    end
+
+    def eof?
+      @cursor >= @table.n_rows
+    end
+
+    def decode(target, **)
+      return if eof?
+
+      rec = Record.new(@table, @cursor)
+      @cursor += 1
+
+      target = target.allocate if target.is_a?(Class)
+      target.from_parquet(rec)
+      target
+    end
+
+    private
+
+    def read_table
```

Comment:
So the big downside of reading Parquet in Ruby that you MUST load the whole file into memory which is :hankey:. Nothing we can do about it, but something to keep in mind.

Reply:
as long as the input data is partitioned sensibly we should be ok

Reply:
Parquet sucks! Is there any reason the devs wouldn't update to add stream support? Is there a something about the implementation that requires it to be in memory?

Reply:
https://issues.apache.org/jira/browse/ARROW-9137